cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactClassLoaderFactory.java

```diff
@@ -73,8 +73,14 @@ CloseableClassLoader createClassLoader(Location artifactLocation) throws IOExcep
       // It is needed because we don't know what program types that an artifact might have.
       // TODO: CDAP-5613. We shouldn't always expose the Spark classes.
       ProgramRunner programRunner = programRunnerFactory.create(ProgramType.SPARK);
-      if (programRunner instanceof ProgramClassLoaderProvider) {
-        programClassLoader = ((ProgramClassLoaderProvider) programRunner).createProgramClassLoader(cConf, unpackDir);
+      try {
+        if (programRunner instanceof ProgramClassLoaderProvider) {
+          programClassLoader = ((ProgramClassLoaderProvider) programRunner).createProgramClassLoader(cConf, unpackDir);
+        }
+      } finally {
+        if (programRunner instanceof Closeable) {
```

Comment:
Oh nvm. I see that the ``SparkProgramRunner`` is made ``Closeable`` later in this commit. I was looking at the release branch and couldn't find it.